### ISSUE TITLE
Buggy header shadow

### ISSUE THREAD

**Issue by PegasisForever** [squidfunk/mkdocs-material#2488]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

Header shadow randomly appears when it shouldn't.
This affects almost every site using mkdocs-material, e.g. https://notes.pegas.is/, https://walkccc.me/LeetCode/
This affects both Chrome and Firefox.
However, the official documentation site doesn't seem to be affected. (probably due to it's announcement bar?)

### Expected behavior

Header shadow only appears when the page is scrolled down.

### Actual behavior

Video: https://cloud.pegasis.site/s/Q5H7LFWoKtWJHcd

### Steps to reproduce the bug

It's so random that I can't give exact steps, try repeating those steps multiple times in different orders:
- resize window
- scroll page
- refresh

### Package versions

* Python: `3.9.2`
* MkDocs: `1.1.2`
* Material: `7.0.6`

### Project configuration

``` yaml
site_name: Pega's Notes
docs_dir: public
site_dir: build
theme:
  name: material
  custom_dir: .overrides
  font:
    text: Fira Sans
    code: JetBrains Mono
  palette:
    primary: teal
    accent: teal
  logo: pega_logo_filled.svg
  favicon: favicon.png
  features:
    - navigation.instant
    - toc.integrate
    - search.suggest
plugins:
  - awesome-pages
  - section-index
  - search:
      lang:
        - en
  - git-revision-date-localized:
      enable_creation_date: true
      timezone: Amarica/Toronto
      locale: en
markdown_extensions:
  - pymdownx.snippets:
      check_paths: true
  - pymdownx.superfences
  - pymdownx.magiclink
  - pymdownx.tilde
  - pymdownx.caret
  - pymdownx.emoji
  - pymdownx.mark
  - pymdownx.smartsymbols
  - toc:
      permalink: true
extra:
  homepage: https://pegas.is/
  disqus: pega-notes
  social:
    - icon: fontawesome/brands/github
      link: https://github.com/PegasisForever
    - icon: fontawesome/brands/instagram
      link: https://www.instagram.com/__pegasis/
extra_css:
  - extra.css
extra_javascript:
  - extra.js
site_url: https://blog.pegas.is/
copyright: Copyright &copy; 2021 Pegasis
```

### System information

* OS: ArchLinux 
* Browser: Google Chrome 89.0.4389.90 & Mozilla Firefox 87.0

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
Thanks for reporting! Unfortunately, I cannot reproduce your issue (on Chrome, macOS). It's all working as intended for me.

- Header shadow is always visible when no tabs are used, e.g. [here](https://squidfunk.github.io/mkdocs-material-example-versioning/latest/)
- When an announcement bar is used, header shadow is visible when scrolling paste it
- When tabs are used, header shadow is visible when scrolling paste them

**Comment by PegasisForever** [comment]
> Header shadow is always visible when no tabs are used, e.g. here

I found the code changing shadow state at [src/assets/javascripts/components/header/_/index.ts](https://github.com/squidfunk/mkdocs-material/blob/master/src/assets/javascripts/components/header/_/index.ts#L183), however I can't find any code checking whether tab are used. Am I missing something?

btw, the shadow in the link you shared also behaves buggy on my computer.

**Comment by squidfunk** [comment]
Thanks for your response. Could you please specify what you mean by "buggy"? Even though the header shadow disappears in your video when the users scrolled to the top, I won't consider it being "buggy". Yes, the behavior might be slightly different depending on browser but everything seems to be behaving predictably.

**Comment by squidfunk** [comment]
Whether the header is shown is determined based on the `Main` area observable:

https://github.com/squidfunk/mkdocs-material/blob/414ffe14cd880eabb234bd6d7ce884a62e0ca7d3/src/assets/javascripts/components/header/_/index.ts#L186-L198

When the `active` is `true`, the header is hidden, which is the case when the main area's top is scrolled past the header's bottom:

https://github.com/squidfunk/mkdocs-material/blob/414ffe14cd880eabb234bd6d7ce884a62e0ca7d3/src/assets/javascripts/components/main/index.ts#L105-L124

**Comment by PegasisForever** [comment]
By "buggy" I mean when the page is not scrolled, sometimes it shows the shadow sometimes it doesn't.

Video: https://cloud.pegasis.site/s/DeS3se7o2Gt53d3

**Comment by squidfunk** [comment]
Okay, thanks for the explanation and the video, I can now reproduce it on the example versioning docs. I'd regard this as an edge case and a very minor bug, but I'll look into it when I find some time.

**Comment by PegasisForever** [comment]
I found the issue. There're actually 2 bugs.
1. https://github.com/squidfunk/mkdocs-material/blob/414ffe14cd880eabb234bd6d7ce884a62e0ca7d3/src/assets/javascripts/components/main/index.ts#L96
    `offsetTop` always return an integer, however the height of the header may not be an integer, this results the calculated scroll position to be something like 0.5(this is fine) or -0.5(not fine!) when scrolled to top.
2. https://github.com/squidfunk/mkdocs-material/blob/414ffe14cd880eabb234bd6d7ce884a62e0ca7d3/src/assets/javascripts/components/main/index.ts#L115
    `<=` should be `<`

I'm working on a pr.

**Comment by squidfunk** [comment]
Fixed in f43ef69ee. I switched all computations to the rounded values, which should now always yield consistent results.

Really good catch!

**Comment by squidfunk** [comment]
Released as part of 7.0.7.
